index.bs

```diff
@@ -1385,38 +1383,49 @@ Build a composed graph up to a given output operand into a computational graph a
+    1. Let |promise| be [=a new promise=].
     1. Run the following steps [=in parallel=]:
-        1. If |outputs| is empty, then [=queue an ML task=] with |global| to [=reject=] |promise| with a {{TypeError}}, and abort these steps.
-        1. [=map/For each=] |name| → |operand| of |outputs|:
-            1. If |name| is empty, then [=queue an ML task=] with |global| to [=reject=] |promise| with a {{TypeError}}, and abort these steps.
-        1. If any of the following sub-steps fail, then [=queue an ML task=] with |global| to [=reject=] |promise| with an "{{OperationError}}" {{DOMException}}, and abort these steps.
-            1. Let |graph| be a new {{MLGraph}} with |realm|:
-                1. Set |graph|.{{MLGraph/[[context]]}} to [=this=].{{MLGraphBuilder/[[context]]}}.
-            1. Make a request to the underlying platform to:
-                1. Connect |graph| to a new [=implementation-defined=] graph implementation |graphImpl| given |graph|.
-                1. Set |graph|.{{MLGraph/[[implementation]]}} to |graphImpl|.
-            1. Make a request to the underlying platform to initialize the graph:
-                1. [=map/For each=] |name| → |operand| of |outputs|:
-                    1. If [=MLOperand/validating MLOperand=] given |operand| and [=this=] returns false, then [=queue an ML task=] with |global| to [=reject=] |promise| with a {{TypeError}}, and abort these steps.
-                    1. If |operand| was created as an input by the underlying platform:
-                        1. If |operand|.{{MLOperand/[[name]]}} is not unique for |graphImpl|, then [=queue an ML task=] with |global| to [=reject=] |promise| with a {{TypeError}}, and abort these steps.
-                        1. Add |operand|.{{MLOperand/[[descriptor]]}} to |graph|.{{MLGraph/[[inputDescriptors]]}}[|operand|.{{MLOperand/[[name]]}}].
-                    1. If |operand| was created as a constant by the underlying platform:
-                        1. Implementations MAY preprocess and optimize the tensor data of |operand| for the underlying platform.
-                    1. Register a [=platform operand=] representing |operand| in |graphImpl| as graph output.
-                    1. Register a [=platform operator=] representing |operand|.{{MLOperand/[[operator]]}} to |graphImpl|.
-
-                Issue(552): Decide how to specify graph initialization.
+        1. Let |graphImpl| be the result of converting [=this=]'s [=MLGraphBuilder/graph=] with |operands|, |operators|, |inputs|, and |outputs|'s [=map/values=] into an [=implementation-defined=] format which can be interpreted by the underlying platform.
+            1. If the underlying platform does not support a requested feature, then [=queue an ML task=] with |global| to [=reject=] |promise| with an "{{OperationError}}" {{DOMException}}, and abort these steps.
+
+            Issue: Reference [=platform operator=] and [=platform operand=] here, or drop those definitions?
```

Comment:
Drawing attention to this note - I'd like to resolve this before merging. As a reminder, this is referencing text in the programming model:

> The `MLGraph` is composed of **platform operators** and **platform operands** which correspond to the operators and _MLOperand_s, but are not script-visible and may be compositions or decompositions of the graph as constructed by script.

My preference would be to replace that text with something like this:

> The `MLGraph` underlying implementation will be composed of platform-specific representations of operators and operands which correspond to the `MLGraphBuilder`'s operators and _MLOperand_s, but which are not script-visible and may be compositions or decompositions of the graph as constructed by script.

... and then just drop this Issue.

Reply:
sounds good to me.

Reply:
Ditto.

Reply:
Filed https://github.com/webmachinelearning/webnn/pull/606 for this.